test/e2e_yaml_test.go

```diff
@@ -0,0 +1,238 @@
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	return
+}
+
+// Substitute docker repos/bucket paths from environment to allow tests on
+// local clusters. Also update namespaces so that examples with ServiceAccounts
+// work under test
+func SubEnv(input []byte, namespace string) ([]byte, error) {
+	val, ok := os.LookupEnv("KO_DOCKER_REPO")
+	var output []byte
+	if ok {
+		re := regexp.MustCompile(DEFAULT_KO_DOCKER_REPO)
+		output = re.ReplaceAll(input, []byte(val))
+	} else {
+		return input, nil
+	}
+
+	re := regexp.MustCompile(DEFAULT_NAMESPACE)
+	output = re.ReplaceAll(output, []byte(strings.ReplaceAll(DEFAULT_NAMESPACE, "default", namespace)))
+	return output, nil
+}
+
+func KoCreate(input []byte, namespace string) ([]byte, error) {
+	cmd := exec.Command("ko", "create", "-n", namespace, "-f", "-")
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return []byte("Cannot create Stdin Pipe"), err
+	}
+
+	go func() {
+		defer stdin.Close()
+		io.WriteString(stdin, string(input))
+	}()
+
+	out, err := cmd.CombinedOutput()
+	return out, err
+}
+
+func DeleteClusterTask(t *testing.T, c *clients, name string) {
+	t.Logf("Deleting clustertask %s", name)
+	err := c.ClusterTaskClient.Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func YamlSubtest(path string, validateFunc func(t *testing.T, c *clients, name string), kind string) func(subt *testing.T) {
+	return func(subt *testing.T) {
+		subt.Parallel()
+
+		// Setup unique namespaces for each test so they can run in complete
+		// isolation
+		c, namespace := setup(subt)
+
+		adminAvailable := true
+
+		// Same check as in github.com/tektoncd/plumbing/scripts/e2e-tests.sh
+		_, err := c.RbacV1Client.ClusterRoleBindings().Get("cluster-admin-binding", metav1.GetOptions{})
+		if err != nil {
+			subt.Logf("cluster-admin-binding result: %s", err)
+			adminAvailable = false
+		}
+
+		// If cluster-admin-binding is available, use it as CI is running, and
+		// admin is required for some examples
+		if adminAvailable {
+			subt.Log("Using cluster-admin user for YAML tests")
+			adminOverride := &clientcmd.ConfigOverrides{
+				Context: api.Context{
+					AuthInfo: "cluster-admin",
+				},
+			}
+
+			c, namespace = setupWithOverrides(subt, adminOverride)
+		}
+
+		knativetest.CleanupOnInterrupt(func() { tearDown(subt, c, namespace) }, subt.Logf)
+		defer tearDown(subt, c, namespace)
+
+		inputExample, err := ioutil.ReadFile(path)
+
+		if err != nil {
+			subt.Fatal(err)
+		}
+
+		subbedInput, err := SubEnv(inputExample, namespace)
+		if err != nil {
+			subt.Fatal(err)
+		}
+
+		out, err := KoCreate(subbedInput, namespace)
+		if err != nil {
+			subt.Fatalf("%s Output: %s", err, out)
+		}
+
+		// Parse from KoCreate for now
+		name, err := GetCreatedTektonCrd(out, kind)
+		if name == "" {
+			// Nothing to check from ko create, this is not a taskrun or pipeline
+			// run
+			subt.Skipf("pipelinerun or taskrun not created")
```

Comment:
can you explain a bit more (maybe in the comment) about what this case is? im a bit confused why we'd be okay with having not created anything, seems like an error?

Reply:
I've added a comment to clarify why some examples are skipped. Below are the affected examples:

```
SKIP: TestExamples/v1alpha1/pipelineruns/task_results_example (3.59s)
SKIP: TestExamples/v1alpha1/taskruns/task-result (3.24s)
SKIP: TestExamples/v1beta1/taskruns/task-result (3.24s)
```

At first, I had thought the examples above intentionally omitted a PipelineRun or TaskRun, however there's a similar `v1beta1/pipelineruns` example that passes (it contains a PipelineRun):

```
PASS: TestExamples/v1beta1/pipelineruns/task_results_example (47.29s)
```

Is it possible the skipped examples above are incomplete?

If so, I can update them in this MR, or create another issue to address these later (at which point I'll remove the option to skip examples missing a TaskRun or PipelineRun).